File: app/src/main/java/com/android/learn/mpresenter/WechatPresenter.java
package com.android.learn.mpresenter;


import com.android.learn.base.mmodel.BaseResponse;
import com.android.learn.base.mmodel.WxArticle;
import com.android.learn.base.mpresenter.BasePresenter;
import com.android.learn.base.thirdframe.rxjava.BaseObserver;
import com.android.learn.mcontract.WechatContract;

import java.util.List;

import io.reactivex.Observable;


public class WechatPresenter extends BasePresenter<WechatContract.View> implements WechatContract.Presenter {

    @Override
    public void getWxArticle() {
        Observable observable = mRestService.getWxArticle();
        addSubscribe(observable, new BaseObserver<BaseResponse<List<WxArticle>>>() {
            @Override
            public void onNext(BaseResponse<List<WxArticle>> datas) {
                mView.showWxArticle(datas.data);
            }
        });
    }

}

File: app/src/main/java/com/android/learn/mpresenter/WechatPresenter.kt
package com.android.learn.mpresenter


import com.android.base.mmodel.BaseListResponse
import com.android.base.mmodel.WxArticle
import com.android.base.mpresenter.BasePresenter
import com.android.base.thirdframe.rxjava.BaseObserver
import com.android.learn.mcontract.WechatContract


class WechatPresenter : BasePresenter<WechatContract.View>(), WechatContract.Presenter {

    override fun getWxArticle() {
        val observable = mRestService.wxArticle
        addSubscribe(observable, object : BaseObserver<BaseListResponse<List<WxArticle>>>(false) {
            override fun onNext(datas: BaseListResponse<List<WxArticle>>) {
                mView!!.showWxArticle(datas.data)
            }
        })
    }

}
